tests/lib.rs

```diff
@@ -876,3 +876,304 @@ Error codes:
         );
     }
 }
+
+#[test]
+fn redact_no_args() {
+    #[derive(FromArgs, Debug)]
+    /// Short description
+    struct Cmd {
+        #[argh(option)]
+        /// a msg param
+        msg: Option<String>,
+    }
+
+    let actual = Cmd::redact(&["<<<arg0>>>"], &[]).unwrap();
```

Comment:
> I don't understand the use of this pattern, <<<arg0>>>. Is this a "magic string" that says, "get the value of arg0 from the environment"?
>
> I don't think that's the case, but it threw me at first, especially because there are a lot of tests (maybe mostly the existing test) that don't return this "<<>>" value in any expected results. (I expected to see some Usage / Help message, for example, that actually echoed back to the test result the <<<arg0>> in the output message.

Great idea on using something different than `<<<arg0>>>`, I'll use hopefully a better string.

> I don't understand why the argument (to redact and to from_args) is an array of strings. If I understand the documentation, these are interpreted as "space-separated". In most cases I would expect the program name to be a single string with no space, and if it's not, but space separated, why not let the user plug in a string with spaces?
>
> That first argument just looks really awkward and uncomfortable to use.

Yeah, I'm not crazy about it either. Unfortunately this is required by how the help string is generated. Subcommands need some notion of what the parent commands were named so it can generate a help string like `"Usage: my-command foo bar <command> [<args>]"`. Perhaps we could come up with a better way of passing down this context, but I think that would make some breaking changes with the API, which is a bit more of an invasive change. `FromArgs::redact()` picks this up from the nature of using the shared functionality. That's why redacting arguments with `--help` will return `Err(EarlyExit { ... })` instead of returning `vec!["foo", "--help"]`.

> If you're adding a new method, what do you think about adding a new version of from_args (new name of course) with better ergonomics, and deprecate from_args()? (I don't dislike the name from_args, but I assume you can't change the signature of the existing method very easily.)

We can consider doing that, but we ought to do that though a breaking change in a separate PR.